node/src/test_utils/recorder_stop_conditions.rs

```diff
@@ -0,0 +1,325 @@
+        predicate(msg as RefMsgExpected)
+    }
+}
+
+#[macro_export]
+macro_rules! single_type_id {
+    ($single_message: ident) => {
+        StopConditions::Single(StopCondition::StopOnType(TypeId::of::<$single_message>()))
+    };
+}
+
+#[macro_export]
+macro_rules! multiple_type_ids{
+    ($($single_message: ident),+) => {
+         StopConditions::All(vec![$(StopCondition::StopOnType(TypeId::of::<$single_message>())),+])
+    }
+}
+
+mod tests {
+    use crate::accountant::{ResponseSkeleton, ScanError, ScanForPayables};
+    use crate::daemon::crash_notification::CrashNotification;
+    use crate::sub_lib::peer_actors::{NewPublicIp, StartMessage};
+    use crate::test_utils::recorder_stop_conditions::{StopCondition, StopConditions};
+    use masq_lib::messages::ScanType;
+    use std::any::TypeId;
+    use std::net::{IpAddr, Ipv4Addr};
+    use std::vec;
+
+    #[test]
+    fn stop_on_match_works() {
+        let mut cond1 = StopConditions::Single(StopCondition::StopOnMatch {
+            exemplar: Box::new(StartMessage {}),
+        });
+        let mut cond2 = StopConditions::Single(StopCondition::StopOnMatch {
+            exemplar: Box::new(NewPublicIp {
+                new_ip: IpAddr::V4(Ipv4Addr::new(1, 8, 6, 4)),
+            }),
+        });
+        let mut cond3 = StopConditions::Single(StopCondition::StopOnMatch {
+            exemplar: Box::new(NewPublicIp {
+                new_ip: IpAddr::V4(Ipv4Addr::new(44, 2, 3, 1)),
+            }),
+        });
+        let tested_msg = NewPublicIp {
+            new_ip: IpAddr::V4(Ipv4Addr::new(44, 2, 3, 1)),
+        };
+
+        assert_eq!(
+            cond1.resolve_stop_conditions::<NewPublicIp>(&tested_msg),
+            false
+        );
+        assert_eq!(
+            cond2.resolve_stop_conditions::<NewPublicIp>(&tested_msg),
+            false
+        );
+        assert_eq!(
+            cond3.resolve_stop_conditions::<NewPublicIp>(&tested_msg),
+            true
+        )
+    }
+
+    #[test]
+    fn stop_on_predicate_works() {
+        let mut cond_set = StopConditions::Single(StopCondition::StopOnPredicate {
+            predicate: Box::new(|msg| {
+                let scan_err_msg: &ScanError = msg.downcast_ref().unwrap();
+                scan_err_msg.scan_type == ScanType::PendingPayables
+            }),
+        });
+        let wrong_msg = ScanError {
+            scan_type: ScanType::Payables,
+            response_skeleton_opt: None,
+            msg: "booga".to_string(),
+        };
+        let good_msg = ScanError {
+            scan_type: ScanType::PendingPayables,
+            response_skeleton_opt: None,
+            msg: "blah".to_string(),
+        };
+
+        assert_eq!(
+            cond_set.resolve_stop_conditions::<ScanError>(&wrong_msg),
+            false
+        );
+        assert_eq!(
+            cond_set.resolve_stop_conditions::<ScanError>(&good_msg),
+            true
+        )
+    }
+
+    #[test]
+    fn match_any_works() {
```

Comment:
Maybe mention something about the number of conditions not decreasing as the conditions are resolved, since this test does assert that. It'd help explain what `inspect_len_of_any` is all about.